src/components/enterprise-user-subsidy/data/hooks.jsx

```diff
@@ -9,55 +9,38 @@ import offersReducer, { initialOfferState } from '../offers/data/reducer';
 
 import { LICENSE_STATUS } from './constants';
 import { fetchSubscriptionLicensesForUser } from './service';
-import {
-  hasValidStartExpirationDates,
-  isDefinedAndNotNull,
-  isDefinedAndNull,
-} from '../../../utils/common';
 import { features } from '../../../config';
 
-export function useSubscriptionLicenseForUser(subscriptionPlan) {
+export function useSubscriptionLicenseForUser(enterpriseId) {
   const [license, setLicense] = useState();
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    if (isDefinedAndNull(subscriptionPlan)) {
-      setIsLoading(false);
-      return;
-    }
-
-    if (isDefinedAndNotNull(subscriptionPlan)) {
-      if (!hasValidStartExpirationDates(subscriptionPlan)) {
-        setIsLoading(false);
-        return;
-      }
-
-      fetchSubscriptionLicensesForUser(subscriptionPlan.uuid)
-        .then((response) => {
-          const { results } = camelCaseObject(response.data);
-          const activated = results.filter(result => result.status === LICENSE_STATUS.ACTIVATED);
-          const assigned = results.filter(result => result.status === LICENSE_STATUS.ASSIGNED);
-          const revoked = results.filter(result => result.status === LICENSE_STATUS.REVOKED);
-
-          if (activated.length) {
-            setLicense(activated.pop());
-          } else if (assigned.length) {
-            setLicense(assigned.pop());
-          } else if (revoked.length) {
-            setLicense(revoked.pop());
-          } else {
-            setLicense(null);
-          }
-        })
-        .catch((error) => {
-          logError(new Error(error));
+    fetchSubscriptionLicensesForUser(enterpriseId)
+      .then((response) => {
+        const { results } = camelCaseObject(response.data);
+        const activated = results.filter(result => result.status === LICENSE_STATUS.ACTIVATED);
+        const assigned = results.filter(result => result.status === LICENSE_STATUS.ASSIGNED);
+        const revoked = results.filter(result => result.status === LICENSE_STATUS.REVOKED);
+
+        if (activated.length) {
+          setLicense(activated.pop());
```

Comment:
The endpoint used here returns results ordered by expiration in descending order, correct?

If that's the case and the learner happens to have 2 activated licenses at once, I believe we'd be picking the one that expires sooner.

Do we actually want to use `.shift()` here to return the first element in the list, instead of the last?

Reply:
Results are ordered first by status ascending (all activated Licenses will be first) and second by expiration date descending (the date furthest out in the future will be first). Great catch, will update.